_docs/structs_in_glu.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Using Structures in Glu
+category: Structure related data using Structs
+---
+
+Structures are a way to group related data together in Glu. They are similar to classes in other languages but are typically used for simpler data structures. In this section, we will cover how to declare and use structures in Glu.
+
+## Declaring a Structure
+
+A structure in Glu is declared using the `struct` keyword followed by the structure's name and its fields. Each field has a name and a type. Here are some examples of structure declarations:
+
+```glu
+struct Person {
+    name: String,
+    age: Int,
+}
+```
+
+In this example, the structure `Person` has two fields: name of type `String` and age of type `Int`. You can also provide default values for the fields:
+
+```glu
+struct Person {
+    name: String = "Unknown",
+    age: Int = 0,
+}
+```
+
+In this example, the fields name and age are initialized to `Unknown` and `0` respectively.
+
+The last comma in the field list is optional, so you can write the following:
+
+```glu
+struct Person {
+    name: String,
+    age: Int
+}
+```
+
+## Declaring an Instance of a Structure
+
+You can declare a variable that contains a structure and initialize it with or without default values.
+
+```glu
+struct Person {
+    name: String = "Unknown",
+    age: Int = 0
+}
+
+func main() {
+    let person1: Person = {}; //  initialisation with default values
+    let person2: Person = { "Alice", 25 }; // initialisation with custom values
+    let person3: Person = { "Bob" }; // initialisation with custom name but with default age
+}
+```
+
+## Accessing Fields of a Structure
+
+You can access the fields of a structure using the dot operator (`.`). For example:
+
+```glu
+struct Person {
+    name: String,
+    age: Int
+}
+
+func main() {
+    let person: Person = { "Alice", 25 };
+    std::print(person.name); // prints "Alice"
+    std::print(person.age); // prints 25
+}
+```
+
+## Packing and Unpacking Structures
+
+Glu allows you to pack structures for memory efficiency using the `@packed` attribute. This feature exists in other programming languages and can be useful for certain applications where memory usage is critical.
+
+```glu
+@packed struct Data {
+    a: Int8,
+    b: Int64,
+}
+```
+
+In this example, the structure Data is packed, meaning its fields are laid out in memory without padding. While this can save memory, it comes with a trade-off: accessing the members of a packed structure can be slower than accessing members of a non-packed structure due to potential misalignment in memory.
```

Comment:
better to explain number of bytes before and after `packed`?

Reply:
i've added the explaination